### PR TITLE
feat: upsteams taskfiles from client, adds info on usage, adds exports/

### PR DESCRIPTION
**Pull request by Gowiem**
## Info

* Upstreams our task files from another client
* Adds `exports/Taskfile.dist.yaml`, which is our main file for sharing taskit around
* Adds info on using the project and how to use `exports/Taskfile.dist.yaml`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,22 @@ Taskit (said Task-kit or TK for short) is the Masterpoint [Taskfile](https://tas
 
 ## Using Taskit
 
-Taskit is built around the idea of being consumed by downstream repos. It is pulled through our standard installation `Taskfile.yaml`
+Taskit is built around the idea of being consumed by downstream repos. It is pulled through a standard Taskfile that you can find in [`exports/Taskfile.dist.yaml`](./exports/Taskfile.dist.yaml).
+
+The procedure to add it to a project is to do the following:
+
+1. Copy `exports/Taskfile.dist.yaml` to your project.
```
Comment (gberenice) [inline README.md:11]: We can add a full command here, e.g.:
```sh
curl -sL https://raw.githubusercontent.com/masterpointio/taskit/feature/shared-task-modules/exports/Taskfile.dist.yaml -o Taskfile.dist.yaml
```

File: README.md [hunk 2]
```diff
@@ -4,7 +4,22 @@ Taskit (said Task-kit or TK for short) is the Masterpoint [Taskfile](https://tas
...
+2. Add the following lines to your project's `.gitignore`:
```
Comment (gberenice) [inline README.md:12]: What do you think about adding `.gitignore` modifications to our `task init` flow?

File: README.md [hunk 3]
```diff
@@ -4,7 +4,22 @@ Taskit (said Task-kit or TK for short) is the Masterpoint [Taskfile](https://tas
...
+
+   ```gitignore
+   .taskit/
+   .task/
+   .snaplet/snapshots/
+   ```
+
+3. Run `task init` to initialize taskit by downloading this repo into your remote repo. Note, `git` is a requirement.
+4. Run `task --list` to list all newly available tasks from taskit.
+5. (Optional) Add a `.env.taskit` file which can include overrides for any variables in taskit.
```
Comment (gberenice) [inline README.md:22]: Probably it's worth adding something like "After this, the setup process for `taskit` is all wrapped up and you're now ready to push all the configuration files to the repository."

File: exports/Taskfile.dist.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  clean:
+    desc: Cleans up the project's Task setup
+    silent: true
+    cmds:
+      - |
+        if [ -d .taskit ]; then
+          rm -rf .taskit/
+        fi
+        mkdir -p .taskit
```
Comment (gberenice) [inline exports/Taskfile.dist.yaml:49]:
```suggestion
      - rm -rf .taskit/
      - mkdir -p .taskit
 ```
To make things more compact. `rm -rf` should not fail if the directory does not exist.

File: lib/aws/Taskfile.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    set: [e]
+    vars:
+      AWS_REGION: us-east-1
+      AWS_COMMAND: '{{.AWS_COMMAND | default "sts get-caller-identity"}}'
+    requires:
+      vars:
+        - MESSAGE
+    cmds:
+      - |
+
```
Comment (gberenice) [inline lib/aws/Taskfile.yaml:17]:
```suggestion
```

File: lib/aws/Taskfile.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+version: "3"
+
+tasks:
+  validate-access:
+    desc: Validate access to AWS for the given command
+    # silent: true
```
Comment (gberenice) [inline lib/aws/Taskfile.yaml:6]:
```suggestion
    silent: true
```
Should be silent?

File: lib/snaplet/Taskfile.yaml [hunk 7]
```diff
@@ -0,0 +1,106 @@
+version: "3"
+
+includes:
+  aws:
+    taskfile: ../aws/Taskfile.yaml
+    aliases: [aws]
```
Comment (gberenice) [inline lib/snaplet/Taskfile.yaml:6]: It's equal to a task name. Is there a benefit I'm missing?

File: lib/snaplet/Taskfile.yaml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      - printf "\n\n🗑️ Captured snapshot at {{.SNAPSHOT_PATH}}"
+
+  upload:
+    desc: Helper for uploading a snaplet snapshot to a given S3 Bucket
+    deps: [validate-snaplet-bucket-access]
+    vars:
+      UPLOAD_AS_LATEST: true
+    requires:
+      vars:
+        - SNAPLET_BUCKET
```
Comment (gberenice) [inline lib/snaplet/Taskfile.yaml:70]: Really minor thing, but something that сaught my attention. There are 2 patterns used in the task for arrays:
```
    requires:
      vars:
        - SNAPLET_SOURCE_DATABASE_URL
  ```
and 
```
  deps: [validate-snaplet-bucket-access]
```
We might want to use one of them to look the code nicely readable.

File: lib/snaplet/Taskfile.yaml [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    cmds:
+      - aws s3 cp --quiet --sse=AES256 --recursive {{.SNAPLET_BUCKET}}/{{.SNAPSHOT_NAME}} {{.SNAPSHOT_PATH}}
+      - printf "\n\n⬇ Snapshot downloaded to {{.SNAPSHOT_PATH}}"
+
+  restore:
+    desc: Restores the given SNAPSHOT_NAME to the SNAPLET_TARGET_DATABASE_URL database.
+    prompt: This will wipe all of the data in the given SNAPLET_TARGET_DATABASE_URL. Are you sure?
```
Comment (gberenice) [inline lib/snaplet/Taskfile.yaml:96]: Using prompt 👍

**Review by gberenice — COMMENTED** [review]
Being picky with my comments, but also pretty excited to see this 😍

**Comment by Gowiem** [comment]
@gberenice that was a great PR review, be sure to do all your PR reviews like that from now on 💯 😁. I worked to address all your feedback. Let me know if you have any additional thoughts!

**Review by gberenice — APPROVED** [review]
> be sure to do all your PR reviews like that from now on
 
@Gowiem haha, will try :)
shipit

**Pull request merged**